tests/python_compiler/transforms/test_xdsl_combine_global_phases.py

```diff
@@ -0,0 +1,264 @@
+                    %arg1x2 = arith.mulf %arg1, %two_1 : f64
+                    // CHECK: scf.yield [[arg12:%.*]] : f64
+                    scf.yield %arg1x2 : f64
+                }
+                // CHECK: [[phi_sum:%.*]] = arith.addf [[ret:%.*]], [[arg0:%.*]] : f64
+                // CHECK: quantum.gphase([[phi_sum:%.*]]) :
+                quantum.gphase %ret
+                // CHECK: [[q1:%.*]] = quantum.custom "RX"() [[q0:%.*]] : !quantum.bit
+                %2 = quantum.custom "RX"() %0 : !quantum.bit
+                return
+            }
+        """
+
+        ctx = Context()
+        # Load scf.Scf dialects to ensure operations (scf.if) in the program str are registered.
+        ctx.load_dialect(scf.Scf)
+        # Load arith.Arith dialects to ensure operations (arith.constant, f64) in the program str are registered.
+        ctx.load_dialect(arith.Arith)
+        ctx.load_dialect(func.Func)
+        ctx.load_dialect(test.Test)
+        ctx.load_dialect(Quantum)
+
+        module = xdsl.parser.Parser(ctx, program).parse_module()
+        pipeline = xdsl.passes.PipelinePass((CombineGlobalPhasesPass(),))
+
+        pipeline.apply(ctx, module)
+        run_filecheck(program, module)
+
+    def test_combinable_ops_in_control_flow_if(self, run_filecheck):
+        """Test that combine global phases in a func without control flows."""
+        program = """
+            func.func @test_func(%cond: i32, %arg0: f64, %arg1: f64) {
+                // CHECK: [[q0:%.*]] = "test.op"() : () -> !quantum.bit
+                %0 = "test.op"() : () -> !quantum.bit
+                // CHECK: [[ret:%.*]] = scf.if [[cond:%.*]] -> (f64) {
+                %ret = scf.if %cond -> (f64) {
+                    // CHECK: [[two0:%.*]] = arith.constant [[two0:2.*]] : f64
+                    %two0 = arith.constant 2 : f64
+                    // CHECK: [[arg02:%.*]] = arith.mulf [[arg0:%.*]], [[two0:%.*]] : f64
+                    %arg02 = arith.mulf %arg0, %two0 : f64
+                    // CHECK: [[t0:%.*]] = "test.op"() : () -> f64
+                    %t0 = "test.op"() : () -> f64
+                    // CHECK: [[phi_sum:%.*]] = arith.addf [[t0:%.*]], [[t0:%.*]] : f64
+                    // CHECK: quantum.gphase([[phi_sum:%.*]])
+                    quantum.gphase %t0
+                    quantum.gphase %t0
+                    // CHECK: scf.yield [[arg02:%.*]] : f64
+                    scf.yield %arg02 : f64
+                    // CHECK: } else {
+                } else {
+                    // CHECK: [[two1:%.*]] = arith.constant [[two1:2.*]] : f64
+                    %two1 = arith.constant 2 : f64
+                    // CHECK: [[arg12:%.*]] = arith.mulf [[arg1:%.*]], [[two1:%.*]] : f64
+                    %arg1x2 = arith.mulf %arg1, %two1 : f64
+                    // CHECK: [[phi_sum:%.*]] = arith.addf [[arg1x2:%.*]], [[arg1x2:%.*]] : f64
+                    // CHECK: quantum.gphase([[phi_sum:%.*]])
```

Comment:
Maybe FileCheck is able to handle this, but I would have thought that defining `phi_sum` multiple times would be a problem.

```suggestion
                    // CHECK: [[phi_sum_x2:%.*]] = arith.addf [[arg1x2:%.*]], [[arg1x2:%.*]] : f64
                    // CHECK: quantum.gphase([[phi_sum_x2:%.*]])
```